### ISSUE TITLE
apply the optimized parameters

### ISSUE THREAD

**Issue by Zoezoehooray** [huggingface/setfit#298]
I did my hyperparameter search optimization on one computer and now I'm trying to apply the obtained parameters on another computer, so I could not use this code "trainer.apply_hyperparameters(best_run.hyperparameters, final_model=True)
trainer.train()". I put the obtained parameters manually in my new trainer instead. But I have two sets of the parameters, and I'm not sure which set to use. For example, in the line above I have seed =9, but below seed = 8.

Here are the obtained parameters from the optimization:

Trial 14 finished with value: 0.8711734693877551 and parameters: {'learning_rate': 1.0472016582222107e-05, 'num_epochs': 1, 'batch_size': 4, 'num_iterations': 40, 'seed': 9, 'max_iter': 54, 'solver': 'lbfgs', 'model_id': 'sentence-transformers/all-mpnet-base-v2'}. Best is trial 14 with value: 0.8711734693877551.
Trial: {'learning_rate': 5.786637612112363e-05, 'num_epochs': 1, 'batch_size': 4, 'num_iterations': 20, 'seed': 8, 'max_iter': 52, 'solver': 'lbfgs', 'model_id': 'sentence-transformers/all-mpnet-base-v2'}
model_head.pkl not found on HuggingFace Hub, initialising classification head with random weights. You should TRAIN this model on a downstream task to use it for predictions and inference.
***** Running training *****
  Num examples = 73160
  Num epochs = 1
  Total optimization steps = 18290
  Total train batch size = 4

Thank you so much!!

**Comment by tomaarsen** [comment]
Hello!

That is a little odd, I did not expect the
```
Trial: {'learning_rate': 5.786637612112363e-05, 'num_epochs': 1, 'batch_size': 4, 'num_iterations': 20, 'seed': 8, 'max_iter': 52, 'solver': 'lbfgs', 'model_id': 'sentence-transformers/all-mpnet-base-v2'}
```
to be printed. However, it looks like Trial 14 is the best one based on these lines:
```
Trial 14 finished with value: 0.8711734693877551 and parameters: {'learning_rate': 1.0472016582222107e-05, 'num_epochs': 1, 'batch_size': 4, 'num_iterations': 40, 'seed': 9, 'max_iter': 54, 'solver': 'lbfgs', 'model_id': 'sentence-transformers/all-mpnet-base-v2'}. Best is trial 14 with value: 0.8711734693877551.
```

So perhaps we can apply those:
```python
model = SetFitModel.from_pretrained(
    "sentence-transformers/all-mpnet-base-v2",
    head_params = {
        "max_iter": 54,
        "solver": "lbfgs",
    }
)

trainer = SetFitTrainer(
    model,
    train_dataset=train_dataset,
    eval_dataset=eval_dataset,
    num_epochs=1,
    num_iterations=40,
    batch_size=4,  # <- You can probably toy with this, e.g. put it higher for slightly higher training speed
    learning_rate=1.047e-05  # <- rounded it a bit for convenience
)
# the `seed` should not matter much

trainer.train()
metrics = trainer.evaluate()

print(metrics)
```

And otherwise you can also try the values from the other Trial and re-train with those, after which you can compare the results. You can get the metrics via `trainer.evaluate()`, which evaluates on `eval_dataset`. 

I'll close this for now, but feel free to reopen if you feel like this issue isn't solved now!

- Tom Aarsen
